src/cleanlab_tlm/utils/rag.py

```diff
@@ -135,6 +140,45 @@ def __init__(
 
         _validate_trustworthy_rag_options(options=options, initialized_evals=self._evals)
 
+        # Optional per-eval tool call overrides
+        # These are name-based include/exclude sets used only in the _handle_tool_call_filtering decorator
+        self._configure_tool_call_eval_overrides(exclude_names=[k.name for k in self._evals if k.response_identifier])
+
+    def _configure_tool_call_eval_overrides(
+        self,
+        *,
+        exclude_names: Optional[list[str]] = None,
+    ) -> None:
+        """Configure per-eval overrides for tool call handling.
+
+        This method allows fine-grained control over which evals are processed vs.
+        skipped (scored as None) when the response is detected as a tool call, without
+        modifying the Eval objects.
+
+        - If an eval name is in exclude_names, it will be filtered (score=None) during tool call handling.
+
+        Args:
+            exclude_names (list[str] | None): Evaluation names to always filter during tool calls.
+                This replaces the existing exclude set.
+        """
+        exclude_names = exclude_names or []
+
+        def _is_valid_name(name: str) -> bool:
+            is_eval_name = any(eval_obj.name == name for eval_obj in self._evals)
+            eval_obj = next((eval_obj for eval_obj in self._evals if eval_obj.name == name), None)
+            has_response_identifier = eval_obj is not None and eval_obj.response_identifier is not None
+            return is_eval_name and has_response_identifier
+
+        # Validate that the exclude names are valid eval names
+        invalid_names = [name for name in exclude_names if not _is_valid_name(name)]
+
+        if invalid_names:
+            raise ValidationError(
+                f"Invalid eval name{'s' if len(invalid_names) > 1 else ''}: {', '.join(invalid_names)}"
+            )
+
+        self._tool_call_eval_exclude_names = set(exclude_names)
```

Comment:
[nit] just curious, is there a reason why the input arg is a list and here we convert to set?

Reply:
I accepted a list in `__init__()` for ergonomics/validation, but store it as a set because it’s conceptually a filter (membership check), not an ordered list. Later, `exclude_names` was removed from the initializer, but I kept the internal set since it still works well for this purpose.